### PR TITLE
jets: fix cut for 0 run

### LINKED ISSUE #69: +cut jet tries to allocate size 0 indirect atom
`(cut [3 [0 0] 0])` causes Ares to crash:

@philipcmonk wrote this I believe?

```
thread '<unnamed>' panicked at 'assertion failed: size > 0', external/crate_index__ares-0.1.0/src/noun.rs:314:9
stack backtrace:
0: _rust_begin_unwind
1: core::panicking::panic_fmt
2: core::panicking::panic
3: ares::noun::IndirectAtom::new_raw_mut
4: ares::noun::IndirectAtom::new_raw_mut_zeroed
5: ares::noun::IndirectAtom::new_raw_mut_bitslice
6: ares::jets::math::jet_cut
7: ares::interpreter::match_pre_hint
8: ares::interpreter::interpret::{{closure}}
...
```

### PR DESCRIPTION
**Pull request by eamsden**
Fixes #69

### CODE REVIEW

**Comment by drbeefsupreme** [comment]
Yup, this works!

File: rust/ares/src/jets/math.rs [hunk 1]
```diff
@@ -543,6 +543,12 @@ pub fn jet_cut(stack: &mut NockStack, subject: Noun) -> Result<Noun, JetErr> {
     let run = raw_slot(arg, 13).as_direct()?.data() as usize;
     let atom = raw_slot(arg, 7).as_atom()?;
 
+    if run == 0 {
+        unsafe {
+            return Ok(DirectAtom::new_unchecked(0).as_noun());
+        }
```
Comment (philipcmonk) [inline rust/ares/src/jets/math.rs:549]: Typically we use the `D` shortcut for this.

```suggestion
        return Ok(D(0));
```

**Review by philipcmonk — CHANGES_REQUESTED** [review]
Thanks for fixing this!

**Review by philipcmonk — APPROVED** [review]

**Review by ashelkovnykov — COMMENTED** [review]
LGTM

**Pull request merged**